libsolidity/formal/Why3Translator.cpp

```diff
@@ -428,7 +428,7 @@ bool Why3Translator::visit(BinaryOperation const& _binaryOperation)
 	Type const& commonType = *_binaryOperation.annotation().commonType;
 	Token::Value const c_op = _binaryOperation.getOperator();
 
-	if (commonType.category() == Type::Category::IntegerConstant)
+	if (commonType.category() == Type::Category::NumberConstant)
```

Comment:
I'm actually not certain how to go about this one. I'm leaving it for now...going to focus on the other parts of this.